File: 🟡 Medium/2530 - Maximal Score After Applying K Operations/_2530MaximalScoreAfterApplyingKOperations.cpp
class Solution {
public:
    long long maxKelements(vector<int>& nums, int k) {
        priority_queue<int> pq;
        for (auto& num : nums) {
            pq.push(num);
        }

        long long res = 0;
        while (k > 0) {
            int maxi = pq.top();
            res += maxi;
            pq.pop();
            pq.push((maxi + 2) / 3);
            k--;
        }

        return res;
    }
};

File: 🟡 Medium/2530 - Maximal Score After Applying K Operations/_2530MaximalScoreAfterApplyingKOperations.java
class Solution {
    public long maxKelements(int[] nums, int k) {
        PriorityQueue<Integer> pq = new PriorityQueue<>((a, b) -> b - a);
        for (int num : nums) {
            pq.offer(num);
        }

        long res = 0;
        while (k > 0) {
            int maxi = pq.poll();
            res += maxi;
            pq.add((maxi + 2) / 3);
            k--;
        }

        return res;

    }
}